src/PairSet.js

```diff
@@ -0,0 +1,17 @@
+export default class PairSet {
+    pairSetMap = new Map();
+
+    add(a, b) {
+        const value = this.pairSetMap.get(a);
+        if (value) {
```

Comment:
since you've already been bitten by it a few too many times, i think now is a good time to start doing explicit `!= null` checks instead of relying on truthiness conversions

Reply:
o fair point ye

Reply:
🐇